owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/sharing/OCShareeViewModel.kt

```diff
@@ -0,0 +1,60 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author David González Verdugo
+ * Copyright (C) 2019 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.presentation.viewmodels.sharing
+
+import android.accounts.Account
+import android.content.Context
+import androidx.lifecycle.ViewModel
+import com.owncloud.android.domain.sharing.sharees.GetShareesUseCase
+import com.owncloud.android.operations.common.OperationType
+import com.owncloud.android.presentation.UIResult
+import com.owncloud.android.ui.errorhandling.ErrorMessageAdapter
+import org.json.JSONObject
+
+class OCShareeViewModel(
+    val context: Context,
+    account: Account,
+    private val getShareesUseCase: GetShareesUseCase = GetShareesUseCase(
+        context,
+        account
+    )
+) : ViewModel() {
+
+    fun getSharees(searchString: String, page: Int, perPage: Int): UIResult<ArrayList<JSONObject>> {
+        getShareesUseCase.execute(
+            GetShareesUseCase.Params(
+                searchString,
+                page,
+                perPage
+            )
+        ).also { useCaseResult ->
```

Comment:
useless usage of also

Reply:
I do not get why is useless here. We execute the operation in the block that returns a result, and we use that result(useCaseResult) to do more things with it. Could you point me to somewhere in the Kotlin documentation where it says it?

Reply:
Because you are not returning useCaseResult. For me, also makes sense to use when you have a result and you want to return it and apply an operation.

For example:

```
fun fetchData(): List<String> =
     remoteDataSource.fetchData().also {
        localDataSource.saveData(it)
    }
```

Here, in your code, you are not used the result anymore.

Reply:
I see your point but I think you can use `also` when you want to apply additional effects to something without the necessity of returning it. From [Kotlin documentation](https://kotlinlang.org/docs/reference/scope-functions.html#also):

![Screenshot 2019-11-06 at 17 11 47](https://user-images.githubusercontent.com/15261363/68315794-9d399300-00b8-11ea-83f2-e4a157edfe44.png)

Reply:
> Usually, you can remove the calls of also from the call chain without breaking the program logic.

If you remove it, you won't convert a DataResult instance into a UIResult.

In this case, either change it by "let" or remove "also" and do it sequentially in order to avoid over-indentation.